### PR TITLE
Fix #9805 - Use timezone offset for datetime only

### LINKED ISSUE #9805: Report Period conditions use timezone offset for Date field types
#### Issue

When a User has set a different timezone and they attempt to report on Date fields, the CRM queries the database using a datetime value with an offset for the timezone. This provides incorrect results as the query will assume a date type has a midnight timestamp.
#### Expected Behavior
When the the user has set a timezone that is not the same as the database timezone.
The Date Period condition will return the correct results for "Today", "Yesterday" etc

#### Actual Behavior

When the the user has set a timezone that is not the same as the database timezone.
Date Period conditions of "Yesterday" will return dates from "Today" if the timezone is behind the db timezone.

#### Possible Fix

If the field type is "datetime" then provide an offset value, else use the midnight value for "date"

#### Steps to Reproduce

1. Set your timezone to America/Chicago in your profile.
2. Create a date field in Contacts
3. Create a Contacts record with the newly created date field set to today.
4. Create a Report on Contacts with a condition on the newly created date field equal to the period "Today"
5. The Contacts record will not be returned in the report

#### Your Environment

* SuiteCRM Version used: SuiteCRM 7.12.7
* Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)): Firefox 105
* Environment name and version (e.g. MySQL, PHP 7): PHP7.4
* Operating System and version (e.g Ubuntu 16.04): Ubuntu 20.0.5

### PR DESCRIPTION
**Pull request by ojs87**
## Description

See https://github.com/salesagility/SuiteCRM/issues/9805

When a User has set a different timezone and they attempt to report on Date fields, the CRM queries the database using a datetime value with an offset for the timezone. This provides incorrect results as the query will assume a date type has a midnight timestamp.

## Motivation and Context
Users with custom timezones may have incorrect report results.

## How To Test This
1. Set your timezone to America/Chicago in your profile.
2. Create a date field in Contacts
3. Create a Contacts record with the newly created date field set to today.
4. Create a Report on Contacts with a condition on the newly created date field equal to the period "Today"
5. The Contacts record will not be returned in the report

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [x] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [ ] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

### CODE REVIEW

**Review by jack7anderson7 — APPROVED** [review]

File: modules/AOR_Reports/aor_utils.php [hunk 1]
```diff
@@ -226,7 +226,7 @@ function getConditionsAsParameters($report, $override = array())
  * @param $date_time_period_list_selected
  * @return DateTime
  */
-function getPeriodDate($date_time_period_list_selected)
+function getPeriodDate($date_time_period_list_selected, $type)
```
Comment (clemente-raposo) [inline modules/AOR_Reports/aor_utils.php:229]: Hi @ojs87, thank you for your PR. Do we want to make `$type` mandatory? should we define it like the following?

```
function getPeriodDate($date_time_period_list_selected, $type = '')
```

Comment (ojs87) [reply modules/AOR_Reports/aor_utils.php]: Hi @clemente-raposo , thanks for your feedback. I've updated the variable to make it non-mandatory as suggested!

**Review by clemente-raposo — DISMISSED** [review]
LGTM

**Comment by johnM2401** [comment]
Hey @ojs87 !

I've had a look and it seems the issue is partly resolved for Today/Yesterday
However, the issue is still present for the other Period Parameters

See:

As an example using "This/Last Year"

This contact has a Date set on the first day of the year:
image


However, they show in both "This Year" and "Last Year" periods:
image

image


<br>
<br>


Additionally, the date value always seems to show the previous day.

<br>

For Example, using "Today", it is set as 2023 - 5th of June
image

But the report displays June 4th for the "Today" period:
image


<br>

Could you have another look, when you get a chance?
Thanks!

**Comment by ojs87** [comment]
Hi John,

I have updated this PR with fixes for the issues you found.
Reports should now display the correct Date on the report results.
The record should now only appear in the correct Period for the date.

**Review by clemente-raposo — APPROVED** [review]

**Pull request merged**